SPIKE: Enable snapstart in dev

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Enable snapstart in dev

### Why did it change

This is a spike to see what happens if we use snapstart. https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html

* It only applies changes in the dev env
* Snapstart only works with x86_64 architecture
* Our step functions weren't invoking our lambdas with a published version (via an alias). Instead they were just invoking whatever the latest version was. You need to specify the version for snapstart to work, hence the changes to the step function resources in the template.
* There needed to be a change to the TokenRequestValidator. The audience value to validate the request against had been baked in to the snapstart image and was set to the wrong thing. This ensures that the audience gets set at first runtime.

These changes made the dev experience much...snappier. It might be a good thing to implement, and if we like it we could maybe even use it in prod and turn off provisioned concurrency. Do we *need* to run arm64 in prod though? We'd also need to do a bunch of checking to ensure we weren't shooting ourselves in the foot like with the audience issue mentioned above.

Wynndow committed:
SPIKE: Enabled snapstart in dev

This is a spike to see what happens if we use snapstart.
https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html

* It only applies changes in the dev env
* Snapstart only works with x86_64 architecture
* Our step functions weren't invoking our lambdas with a published
  version (via an alias). Instead they were just invoking whatever the
latest version was. You need to specify the version for snapstart to
work, hence the changes to the step function resources in the template.
* There needed to be a change to the TokenRequestValidator. The audience
  value to validate the request against had been baked in to the
snapstart image and was set to the wrong thing. This ensures that the
audience gets set at first runtime.

These changes made the dev experience much...snappier. It might be a
good thing to implement, and if we like it we could maybe even use it in
prod and turn off provisioned concurrency. Do we *need* to run arm64 in
prod though? We'd also need to do a bunch of checking to ensure we
weren't shooting ourselves in the foot like with the audience issue
mentioned above.

Wynndow committed:
SPIKE: Remove calls to parameter store from init

In a few places we were fetching params from SSM in the init of the
lambdas. This was an issue in dev when deploying new stacks from scratch
as it baked in the wrong parameter values to the snapstart image. It
would also mean that the param couldn't be changed until a new version
of the lambda was created so a new snapstart image created.

This also fixes up some stuff around the component ID. We are using a
log helper which was just setting a static value of `core` instead of
the actual component ID from the config.

blakeyp commented on lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java:
```diff
@@ -340,7 +337,7 @@ private AuditEvent buildProfileMatchedAuditEvent(
         return new AuditEvent(
                 AuditEventTypes.IPV_GPG45_PROFILE_MATCHED,
-                componentId,
+                configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
```
Is there a cost to these repeated calls to SSM? I guess not because it's cached?

Wynndow replied:
Yep - it'll be slightly more than what we were doing before (storing the component ID on init), but a drop in the ocean in terms of SSM calls.

Wynndow replied:
But yeah, we have the three minute cache so it's not like there's going to be loads more.

Wynndow committed:
BAU: Move Architecture to globals

We're probably not going to start running different lambdas on different
arch's, so we can move this to the globals section to simplify.
